addons/binding/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/discovery/GroheOndusDiscoveryService.java

```diff
@@ -0,0 +1,103 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.grohe.ondus.api.OndusService;
+import org.grohe.ondus.api.model.Appliance;
+import org.grohe.ondus.api.model.Location;
+import org.grohe.ondus.api.model.Room;
+import org.openhab.binding.groheondus.internal.handler.AccountBridgeGroheOndusHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Florian Schmidt - Initial contribution
+ */
+public class GroheOndusDiscoveryService extends AbstractDiscoveryService {
+    private static final String PROPERTY_APPLIANCE_ID = "applianceId";
+
+    private static final String PROPERTY_ROOM_ID = "roomId";
+
+    private static final String PROPERTY_LOCATION_ID = "locationId";
+
+    private final Logger logger = LoggerFactory.getLogger(GroheOndusDiscoveryService.class);
+
+    private final AccountBridgeGroheOndusHandler bridgeHandler;
+
+    public GroheOndusDiscoveryService(AccountBridgeGroheOndusHandler bridgeHandler) {
+        super(Collections.singleton(THING_TYPE_APPLIANCE), 30);
+        logger.debug("initialize discovery service");
+        this.bridgeHandler = bridgeHandler;
+        if (bridgeHandler == null) {
+            logger.warn("no bridge handler for scan given");
+        }
+        this.activate(null);
+    }
+
+    @Override
+    protected void startScan() {
+        if (bridgeHandler == null) {
+            return;
+        }
+        OndusService service;
+        try {
+            service = bridgeHandler.getService();
+        } catch (IllegalStateException e) {
+            logger.debug("No instance of OndusService given.", e);
+            return;
+        }
+        List<Appliance> discoveredAppliances = new ArrayList<>();
+        try {
+            discoveredAppliances = getApplianceList(service);
+        } catch (IOException e) {
+            logger.debug("Could not discover appliances.", e);
+            return;
+        }
+
+        discoveredAppliances.forEach(appliance -> {
+            ThingUID bridgeUID = bridgeHandler.getThing().getUID();
+            ThingUID thingUID = new ThingUID(THING_TYPE_APPLIANCE, appliance.getApplianceId());
+
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(PROPERTY_LOCATION_ID, appliance.getRoom().getLocation().getId());
+            properties.put(PROPERTY_ROOM_ID, appliance.getRoom().getId());
+            properties.put(PROPERTY_APPLIANCE_ID, appliance.getApplianceId());
+
+            DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withProperties(properties)
```

Comment:
Since the appliance ID is an UUID I think it is unique over all appliances?

If so, you can use that UUID as `representationProperty` so ESH ignores the result when is an existing thing or discovery result, see: https://www.eclipse.org/smarthome/documentation/development/bindings/discovery-services.html

So if possible add `.withRepresentationProperty(PROPERTY_APPLIANCE_ID)` here and also `<representation-property>applianceId</representation-property>` to the thing type defintion.

Reply:
Nice, done :)